docs/08-upgrading/06-upgrade-to-three.md

```diff
@@ -0,0 +1,226 @@
+# Upgrade to 3.0
+
+## Web Server: Widget to Component Rename
+
+In Serverpod 3.0, all web server related "Widget" classes have been renamed to "Component" to better reflect their purpose and avoid confusion with Flutter widgets.
+
+The following classes have been renamed:
+
+| Old Name         | New Name            |
+| ---------------- | ------------------- |
+| `Widget`         | `Component`         |
+| `AbstractWidget` | `AbstractComponent` |
+| `WidgetRoute`    | `ComponentRoute`    |
+| `WidgetJson`     | `JsonComponent`     |
+| `WidgetRedirect` | `RedirectComponent` |
+| `WidgetList`     | `ListComponent`     |
+
+If you're importing from the relic library, the import path has changed:
+
+```dart
+// Before
+import 'package:serverpod/src/relic/web_widget.dart';
```

Comment:
Would any consumer really use these `src/` paths?

If so, I think we should add a proper export, as most projects have probably a lint against src imports.